### DISCUSSION TITLE
Defining an upstream dependency for a dbt seed

### DISCUSSION THREAD

**Discussion by TerryLines** [dagster-io/dagster#25265] [Q&A]
I have a situation where one of my operations restores a database from a backup file, so I would like all dbt assets to be downstream of this. @dbt_assets doesn't appear to support dependencies, however it's straightforward to add the dbt sources as outputs when running the op, which more or less solves this, except for seeds. Is there any way to define the seed as a downstream asset, using dbt.cli to materialize it outside of a dbt_asset context?

**Comment by maximearmstrong** [comment]
Hi @TerryLines - I think [this guide](https://docs.dagster.io/integrations/dbt/reference#defining-dependencies) about setting upstream and downstream dependencies for dbt would be helpful.

> Is there any way to define the seed as a downstream asset, using dbt.cli to materialize it outside of a dbt_asset context?

Can you provide more information about your lineage? The goal would be to recreate the file, then run the dbt models?

**Comment by TerryLines** [comment]
I don’t believe that guide covers the possibility of needing to run a seed op? The lineage is that once an op has recreated the database, the seed asset needs to be rerun to create a table in order to run the downstream models. As it stands I have to use a workaround job rather than being able to directly define the seed as downstream.

Sent from my iPhone

On Nov 22, 2024, at 2:06 PM, Maxime Armstrong ***@***.***> wrote:

﻿

Hi @TerryLines<https://github.com/TerryLines> - I think this guide<https://docs.dagster.io/integrations/dbt/reference#defining-dependencies> about setting upstream and downstream dependencies for dbt would be helpful.

Is there any way to define the seed as a downstream asset, using dbt.cli to materialize it outside of a dbt_asset context?

Can you provide more information about your lineage? The goal would be to recreate the file, then run the dbt models?
